include/exadg/postprocessor/pointwise_output_generator_base.cpp

```diff
@@ -280,21 +282,28 @@ PointwiseOutputGeneratorBase<dim, VectorType>::write_time(double time)
 template<typename Number>
 using VectorType = dealii::LinearAlgebra::distributed::Vector<Number>;
 
-template class PointwiseOutputGeneratorBase<2, VectorType<float>>;
-template class PointwiseOutputGeneratorBase<2, VectorType<double>>;
+template class PointwiseOutputGeneratorBase<2, VectorType<float>, float>;
+template class PointwiseOutputGeneratorBase<2, VectorType<double>, double>;
 
-template class PointwiseOutputGeneratorBase<3, VectorType<float>>;
-template class PointwiseOutputGeneratorBase<3, VectorType<double>>;
+template class PointwiseOutputGeneratorBase<3, VectorType<float>, float>;
+template class PointwiseOutputGeneratorBase<3, VectorType<double>, double>;
 
 // BlockVector Instantiations
 template<typename Number>
 using BlockVectorType = dealii::LinearAlgebra::distributed::BlockVector<Number>;
 
-template class PointwiseOutputGeneratorBase<2, BlockVectorType<float>>;
-template class PointwiseOutputGeneratorBase<2, BlockVectorType<double>>;
+template class PointwiseOutputGeneratorBase<2, BlockVectorType<float>, float>;
+template class PointwiseOutputGeneratorBase<2, BlockVectorType<double>, double>;
 
-template class PointwiseOutputGeneratorBase<3, BlockVectorType<float>>;
-template class PointwiseOutputGeneratorBase<3, BlockVectorType<double>>;
+template class PointwiseOutputGeneratorBase<3, BlockVectorType<float>, float>;
+template class PointwiseOutputGeneratorBase<3, BlockVectorType<double>, double>;
+
+// Instantiations for arbitrary collection of pointers to VectorTypes
+template class PointwiseOutputGeneratorBase<2, std::vector<VectorType<float> const *>, float>;
+template class PointwiseOutputGeneratorBase<2, std::vector<VectorType<double> const *>, double>;
+
+template class PointwiseOutputGeneratorBase<3, std::vector<VectorType<float> const *>, float>;
+template class PointwiseOutputGeneratorBase<3, std::vector<VectorType<double> const *>, double>;
```

Comment:
I consider it impractical that different vector versions show up as different instantiations of this class. It looks as if only the `evaluate()` member function depends on `VectorType`, so only this member function should be templated.

Reply:
It's sadly not possible to template `evaluate()` with the current design, since `evaluate()` is virtual.

Reply:
ok. I still believe we should strive for a solution where the vector type is not a template. I'll make another comment elsewhere.